theme/scripts/fadeIn.ts

```diff
@@ -0,0 +1,57 @@
+import isInViewport from './utils/isInViewport';
+import throttle from 'lodash/throttle';
+
+export default (function() {
+  const FadeIn = {
+    SELECTORS: {
+      gridItem: "[data-grid-item]",
+    },
+    ATTRIBUTES: {
+      fadeIn: "data-fade-in",
+    },
+
+    init() {
+      const gridItems = [].slice.call(
+        document.querySelectorAll(FadeIn.SELECTORS.gridItem)
+      );
+
+      if (!gridItems) {
```

Comment:
No need for an else here, you can just do.

```
if (!logic) return; // Exits method
FadeIn.bindEventListerners();
```

You will also want to check for `!gridItems.length` since grid items will always return an empty array

Reply:
Ahh good catch, TY!